tests/common/plugins/sanity_check/checks.py

```diff
@@ -483,8 +483,79 @@ def _check_single_intf_status(intf_status, expected_side):
+    dut_upper_tor = duthosts[0]
+    dut_lower_tor = duthosts[1]
+
+    # Run "show mux status" on dualtor DUTs to collect mux status
+    duts_mux_status = duthosts.show_and_parse('show mux status')
+
+    # Parse and basic check
+    duts_parsed_mux_status = {}
+    for dut_hostname, dut_mux_status in duts_mux_status.items():
+
+        logger.info('Verify that "show mux status" has output ON {}'.format(dut_hostname))
+        if len(dut_mux_status) == 0:
+            err_msg = 'No mux status in output of "show mux status"'
+            return False, err_msg, {}
+
+        logger.info('Verify that mux ports match vlan interfaces of DUT.')
+        vlan_intf_names = set()
+        for vlan in duts_minigraph_facts[dut_hostname]['minigraph_vlans'].values():
+            vlan_intf_names = vlan_intf_names.union(set(vlan['members']))
+        dut_mux_intfs = []
+        for row in dut_mux_status:
+            dut_mux_intfs.append(row['port'])
+        if vlan_intf_names != set(dut_mux_intfs):
+            err_msg = 'Mux ports mismatch vlan interfaces, please check output of "show mux status"'
+            return False, err_msg, {}
+
+        logger.info('Verify mux status and parse active/standby side')
+        dut_parsed_mux_status = {}
+        for row in dut_mux_status:
+            # Verify that mux status is either active or standby
+            if row['status'] not in ['active', 'standby']:
+                err_msg = 'Unexpected mux status "{}", please check output of "show mux status"'.format(row['status'])
+                return False, err_msg, {}
+
+            # Parse mux status, transform port name to port index, which is also mux index
+            port_name = row['port']
+            port_idx = duts_minigraph_facts[dut_hostname]['minigraph_port_indices'][port_name]
+
+            # Transform "active" and "standby" to active side which is "upper_tor" or "lower_tor"
+            status = row['status']
+            if dut_hostname == dut_upper_tor.hostname:
+                # On upper tor, mux status "active" means that active side of mux is upper_tor
+                # mux status "standby" means that active side of mux is lower_tor
+                active_side = UPPER_TOR if status == 'active' else LOWER_TOR
+            else:
+                # On lower tor, mux status "active" means that active side of mux is lower_tor
+                # mux status "standby" means that active side of mux is upper_tor
+                active_side = UPPER_TOR if status == 'standby' else LOWER_TOR
+            dut_parsed_mux_status[str(port_idx)] = active_side
+        duts_parsed_mux_status[dut_hostname] = dut_parsed_mux_status
+
+    logger.info('Verify that the mux status on both ToRs are consistent')
+    upper_tor_mux_status = duts_parsed_mux_status[dut_upper_tor.hostname]
+    lower_tor_mux_status = duts_parsed_mux_status[dut_lower_tor.hostname]
+
+    logger.info('Verify that dualtor hosts have the same set of mux cables')
```

Comment:
If we already check the mux statuses against the VLAN intfs, is this necessary?

Reply:
Removed.